Taxonomy Manager: show a tooltip on taxonomy items count that explains the number

alisterscott opened the pull request:
This PR adds tooltips to the post count on taxonomy items when managing terms.

**Before:**

![before](https://cloud.githubusercontent.com/assets/128826/20513851/661765be-b0d4-11e6-98d1-de3b72a3aeca.gif)

**After:**

![tooltips](https://cloud.githubusercontent.com/assets/128826/20551329/e9f69b32-b18a-11e6-8215-fa46b8bcbd91.gif)

youknowriad commented on client/blocks/taxonomy-manager/list-item.jsx:
```diff
@@ -55,6 +55,12 @@ class TaxonomyManagerListItem extends Component {
+		const { postCount, name, translate } = this.props;
+		const suffix = postCount === 1 ? translate( 'post' ) : translate( 'posts' );
+		return `${postCount} ${name} ${suffix}`;
+	};
```
Shouldn't we use "Pluralization" here? something like:

```
    translate(
        '%(postCount)d %{name}s post',
        '%(postCount)d %{name}s posts',
        {
            count: postCount,
            args: {
                postCount,
                name
            }
        }
    );
```

mtias commented:
I suggest using https://wpcalypso.wordpress.com/devdocs/design/tooltip for this.

timmyc commented:
+1 on @mtias comment above, but big thanks for running with this @alisterscott !

timmyc commented:
On that note, would it be worthwhile to add a `tooltip` prop to the `<Count />` component?

alisterscott commented:
Thanks for all the feedback

@youknowriad: I have used the pluralisation function as suggested for translate, thanks
@mtias: I have switched this over to use the `tooltip` component - see new screenshot above
@timmyc: I have put the `tooltip` component into the `count` component - but I've created a problem as ever since I added the import statement to the tooltip component there's an error message displayed about SSR:

```
PragmaChecker: components/count, dependency components/tooltip is not /** @ssr-ready **/
PragmaChecker: Server Side Rendering constraints not met. Please see https://github.com/Automattic/wp-calypso/tree/master/docs/server-side-rendering.md for details.
```

I'm still starting out with React and don't know a lot about SSR would you have any suggestions on how I could make this `count` component still SSR ready whilst being able to import and use the `tooltip` component?

Functionally it is now working fine it's just this new issue I have introduced.

Thanks!

timmyc commented:
Nice @alisterscott - hmm sorry about sending you down a path that lead to the SSR stuff.  Perhaps we can back that out and just use the tooltip in the `<ListItem />` directly.  Maybe the addition of the tooltip could be hashed out in a subsequent PR.

jordwest commented on client/blocks/taxonomy-manager/list-item.jsx:
```diff
@@ -57,6 +59,29 @@ class TaxonomyManagerListItem extends Component {
+		);
+	};
+
+	showTooltip= () => {
```
Should be a space between `showTooltip` and `=`

jordwest commented on client/blocks/taxonomy-manager/list-item.jsx:
```diff
@@ -57,6 +59,29 @@ class TaxonomyManagerListItem extends Component {
+		this.setState( { showTooltip: true } );
+	};
+
+	hideTooltip= () => {
```
Spacing as above

youknowriad commented on client/components/count/index.jsx:
```diff
@@ -15,8 +16,9 @@ export default React.createClass( {
 	},
 
 	render() {
+		const inheritProps = omit( this.props, [ 'count' ] );
```
Maybe we could write this like that: `const { count, ...inheritProps } = props`

alisterscott replied:
Thanks, updated it

youknowriad commented on client/blocks/taxonomy-manager/list-item.jsx:
```diff
@@ -76,7 +101,20 @@ class TaxonomyManagerListItem extends Component {
+					context={ this.refs && this.refs.count }
+					isVisible={ this.state.showTooltip }
+					position="left"
+					onClose={ noop }
```
I think this prop is useless

alisterscott replied:
Thanks, removed it

jordwest commented on client/blocks/taxonomy-manager/list-item.jsx:
```diff
@@ -105,9 +105,13 @@ class TaxonomyManagerListItem extends Component {
 		this.setState( { showTooltip: false } );
 	};
 
+	name = () => {
```
`name` isn't technically a reserved word, but it is a property of functions in most JS implementations. I doubt this would break anything since it's only defined on the instance, but something like `getName` or `getTermName` might avoid confusion.

alisterscott replied:
Thanks, updated.
